base/cpuid.cpp

```diff
@@ -0,0 +1,63 @@
+#include "base/cpuid.hpp"
+
+#include <string>
+
+#include "base/macros.hpp"
+#if PRINCIPIA_COMPILER_MSVC
+#include <intrin.h>
+#else
+#include <cpuid.h>
+#endif
+
+namespace principia {
+namespace base {
+namespace internal_cpuid {
+
+struct CPUIDResult {
+  std::uint32_t eax;
+  std::uint32_t ebx;
+  std::uint32_t ecx;
+  std::uint32_t edx;
+};
+
+#if PRINCIPIA_COMPILER_MSVC
+CPUIDResult CPUID(const std::uint32_t eax, const std::uint32_t ecx) {
```

Comment:
Could we move the declaration out of the conditional compilation, to make clear that it's common?